lib/admin/components/ServerSettings.js

```diff
@@ -0,0 +1,269 @@
+// @flow
+
+import Icon from '@conveyal/woonerf/components/icon'
+import objectPath from 'object-path'
+import React, {Component} from 'react'
+import {connect} from 'react-redux'
+import Select from 'react-select'
+import {
+  Row,
+  Col,
+  Button,
+  Panel,
+  Glyphicon,
+  FormGroup,
+  ControlLabel
+} from 'react-bootstrap'
+import update from 'react-addons-update'
+
+import { getComponentMessages } from '../../common/util/config'
+import * as adminActions from '../actions/admin'
+import CollapsiblePanel from '../../manager/components/CollapsiblePanel'
+import { FIELDS, SERVER_FIELDS, UPDATER_FIELDS } from '../../manager/util/deployment'
+
+import type { OtpServer, Project } from '../../types'
+import type { AppState, AdminServersState } from '../../types/reducers'
+
+type ContainerProps = {editDisabled: boolean}
```

Comment:
I think this should be defined or inlined down where it's used.

Reply:
It's actually used in the line directly below (in addition to the very bottom of the file).

Reply:
A, makes sense. I recall importing the container props from another container file, but now I see that the redux connection is in this file also, so this makes sense.